File: docs-site/webjs/sdk_onboarding_sub.js
import { CONNECTED, Realtime } from "relayx-webjs"

const client = new Realtime({
  api_key: '$api_key',
  secret: '$secret'
})

client.init()

if(process.argv.length < 3){
    console.log("Usage: node <file>.js <topic>");
    console.log("type = pub or sub")
    process.exit();
}

var topic = process.argv[2]

client.connect()

await client.on(CONNECTED, async (status) => {
    console.log("Connected to relayX!")

    console.log(`Listening for messages on topic => ${topic}`)
    console.log()

    await client.on(topic, (data) => {
        console.log("We got a message!")
        console.log(data)

        console.log()
        console.log("Exiting script...")

        process.exit(0)
    });
})

 setTimeout(async () => {
    await client.close()
    process.exit(0)
}, 15000)

File: docs-site/python/sdk_onboarding_sub.py
import asyncio
import sys
from relayx_py import Realtime

client = Realtime({
  "api_key": '$api_key',
  "secret": '$secret'
})

client.init()

def callback_fn(data):
    print("We got a message!")
    print(data)

    print()
    print("Ctrl + C to quit")

async def on_connected():
    topic = sys.argv[1]

    print("Connected to relayX!")

    print("Listening for messages on topic => {}".format(topic))

    await client.on(topic, callback_fn)

async def main():
    if len(sys.argv) < 3:
        print("Usage: python <file name>.py <topic>")

        print()
        print("Ctrl + C to quit")

    topic = sys.argv[1]

    print("Topic => {}".format(topic))

    await client.on(Realtime.CONNECTED, on_connected)

    await client.connect() 

# Run the main function
if __name__ == "__main__":
    asyncio.run(main())